backup database function broken

Maby your PR will help, but can't test this as the backup database function seems to be broken at API level 21

_Originally posted by @fm-sys in https://github.com/cgeo/cgeo/issues/8786#issuecomment-674792586_

Error Message:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: cgeo.geocaching, PID: 2887
    android.view.InflateException: Binary XML file line #103: Error inflating class <unknown>
        at android.view.LayoutInflater.createView(LayoutInflater.java:633)
        at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:365)
        at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:414)
        at com.android.internal.app.AlertController.installContent(AlertController.java:238)
        at android.app.AlertDialog.onCreate(AlertDialog.java:356)
        at android.app.ProgressDialog.onCreate(ProgressDialog.java:198)
        at android.app.Dialog.dispatchOnCreate(Dialog.java:387)
        at android.app.Dialog.show(Dialog.java:288)
        at android.app.ProgressDialog.show(ProgressDialog.java:116)
        at android.app.ProgressDialog.show(ProgressDialog.java:104)
        at cgeo.geocaching.utils.DatabaseBackupUtils.createBackupInternal(DatabaseBackupUtils.java:85)
        at cgeo.geocaching.utils.DatabaseBackupUtils.lambda$createBackup$3(DatabaseBackupUtils.java:78)
        at cgeo.geocaching.utils.-$$Lambda$DatabaseBackupUtils$nrI7aAGakJoz4qs25vfQpIxtyY8.onClick(lambda)
        at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:165)
        at android.os.Handler.dispatchMessage(Handler.java:111)
        at android.os.Looper.loop(Looper.java:194)
        at android.app.ActivityThread.main(ActivityThread.java:5576)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:956)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:751)
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
        at android.view.LayoutInflater.createView(LayoutInflater.java:607)
        at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55) 
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:365) 
        at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:414) 
        at com.android.internal.app.AlertController.installContent(AlertController.java:238) 
        at android.app.AlertDialog.onCreate(AlertDialog.java:356) 
        at android.app.ProgressDialog.onCreate(ProgressDialog.java:198) 
        at android.app.Dialog.dispatchOnCreate(Dialog.java:387) 
        at android.app.Dialog.show(Dialog.java:288) 
        at android.app.ProgressDialog.show(ProgressDialog.java:116) 
        at android.app.ProgressDialog.show(ProgressDialog.java:104) 
        at cgeo.geocaching.utils.DatabaseBackupUtils.createBackupInternal(DatabaseBackupUtils.java:85) 
        at cgeo.geocaching.utils.DatabaseBackupUtils.lambda$createBackup$3(DatabaseBackupUtils.java:78) 
        at cgeo.geocaching.utils.-$$Lambda$DatabaseBackupUtils$nrI7aAGakJoz4qs25vfQpIxtyY8.onClick(lambda) 
        at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:165) 
        at android.os.Handler.dispatchMessage(Handler.java:111) 
        at android.os.Looper.loop(Looper.java:194) 
        at android.app.ActivityThread.main(ActivityThread.java:5576) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:956) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:751) 
     Caused by: java.lang.RuntimeException: Failed to resolve attribute at index 5
        at android.content.res.TypedArray.getColorStateList(TypedArray.java:425)
        at android.widget.TextView.<init>(TextView.java:1008)
        at android.widget.Button.<init>(Button.java:111)
        at android.widget.Button.<init>(Button.java:107)
        at android.widget.Button.<init>(Button.java:103)
        at java.lang.reflect.Constructor.newInstance(Native Method) 
        at java.lang.reflect.Constructor.newInstance(Constructor.java:288) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:607) 
        at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55) 
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:365) 
        at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:414) 
        at com.android.internal.app.AlertController.installContent(AlertController.java:238) 
        at android.app.AlertDialog.onCreate(AlertDialog.java:356) 
        at android.app.ProgressDialog.onCreate(ProgressDialog.java:198) 
        at android.app.Dialog.dispatchOnCreate(Dialog.java:387) 
        at android.app.Dialog.show(Dialog.java:288) 
        at android.app.ProgressDialog.show(ProgressDialog.java:116) 
        at android.app.ProgressDialog.show(ProgressDialog.java:104) 
        at cgeo.geocaching.utils.DatabaseBackupUtils.createBackupInternal(DatabaseBackupUtils.java:85) 
        at cgeo.geocaching.utils.DatabaseBackupUtils.lambda$createBackup$3(DatabaseBackupUtils.java:78) 
        at cgeo.geocaching.utils.-$$Lambda$DatabaseBackupUtils$nrI7aAGakJoz4qs25vfQpIxtyY8.onClick(lambda) 
        at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:165) 
        at android.os.Handler.dispatchMessage(Handler.java:111) 
        at android.os.Looper.loop(Looper.java:194) 
        at android.app.ActivityThread.main(ActivityThread.java:5576) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:956) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:751) 

```

--- a/main/res/values/themes.xml
+++ b/main/res/values/themes.xml
@@ -225,6 +225,7 @@
     </style>
 
     <style name="Dialog_Alert" parent="@style/Theme.AppCompat.Dialog.Alert">
+        <item name="text_color">@color/text_dark</item>
         <item name="buttonBarButtonStyle">@style/button_borderless</item>
         <item name="background">?android:attr/selectableItemBackground</item>
         <item name="colorAccent">@color/colorAccent</item>
@@ -232,6 +233,7 @@
     </style>
 
     <style name="Dialog_Alert_light" parent="@style/Theme.AppCompat.Light.Dialog.Alert">
+        <item name="text_color">@color/text_light</item>
         <item name="buttonBarButtonStyle">@style/button_borderless</item>
         <item name="background">?android:attr/selectableItemBackground</item>
         <item name="colorAccent">@color/colorAccent</item>

